packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -564,7 +564,7 @@ class HotRunner extends ResidentRunner {
         }
         isolateNotifications.add(
           device.vmService.onIsolateEvent.firstWhere((vm_service.Event event) {
-            return event.kind == vm_service.EventKind.kIsolateRunnable;
+            return event.kind == vm_service.EventKind.kServiceExtensionAdded;
```

Comment:
This is the fix, it turns out the isolate runnable event was coming from the compute isolate, and removing it caused the test to hang. Instead we can use the point where we start registering package:flutter service extensions as an indication that the hot restart is complete

Reply:
Yes, this starts listening too late to pick up the main isolate runnable. I think this is OK to switch to the extension event, because it more accurately measures when we could reasonably perform another hot reload ( though If I see a large change in the benchmarks I can investigate)

Reply:
This is rather unfortunate. I think we do want to ensure we subscribe to the vm service events before we restart isolate. At a minimum let's add a comment/todo explaining why we are listening for ServiceExtensionAdded instead of IsolateRunnable.

Reply:
Done